src/Microsoft.AspNetCore.WebHooks.Receivers/WebHookReceiverConfig.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebHooks.Properties;
+using Microsoft.AspNetCore.WebHooks.Utilities;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.WebHooks
+{
+    /// <summary>
+    /// Provides a default <see cref="IWebHookReceiverConfig"/> implementation which manages
+    /// <see cref="IWebHookReceiver"/> configuration using application settings. The name of the application setting is
+    /// '<c>MS_WebHookReceiverSecret_&lt;name&gt;</c>' where '<c>name</c>' is the name of the receiver, for example
+    /// <c>github</c>. The value is a comma-separated list of secrets, using an ID to differentiate between them. For
+    /// example, '<c>secret0, id1=secret1, id2=secret2</c>'. The corresponding WebHook URI is of the form
+    /// '<c>https://&lt;host&gt;/api/webhooks/incoming/custom/{id}</c>'.
+    /// </summary>
+    public class WebHookReceiverConfig : IWebHookReceiverConfig
```

Comment:
>WebHookReceiverConfig [](start = 17, length = 21)

What isn't this just `IConfiguration`. It sounds like there's a feature here that needs to be thought out

Reply:
The receivers do contain direct references to the `IConfiguration`. But, this is about specially-formatted configuration entries for secret keys. This uses the same key formats as the Microsoft.AspNet.WebHooks version.

Reply:
As discussed offline, I'll switch to `IConfiguration` extension methods supporting a consistent hierarchy of configuration sections e.g.
``` json
"WebHooks": {
  "GitHub": {
    "SecretKey": {
      "default": "default secret key",
      "id1": "secret key for id1"
    }
  }
}
```
I don't think I'll make `"default"` a reserved word because nothing breaks when a URL contains that id. Instead `https://{host}/api/webhooks/incoming/github` and `https://{host}/api/webhooks/incoming/github/default` will use the same secret key.

The other option looks like
``` json
"WebHooks": {
  "GitHub": {
    "SecretKey": {
      "": "default secret key",
      "id1": "secret key for id1"
    }
  }
}
```
That is valid JSON but looks very strange. And I'm not sure our configuration system supports empty property names.